src/Catalog/Helpers/ParallelExecutionUtility.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    public static class ParallelExecutionUtility
+    {
+        public const int DefaultDegreeOfParallelism = 32;
+
+        /// <summary>
+        /// Runs a task created by <paramref name="taskFactory"/> on a number of threads specified by <paramref name="degreeOfParallelism"/> and waits for all tasks to complete.
```

Comment:
>umber of threads spe [](start = 72, length = 20)

This code creates "degreeOfParallelism" number of tasks, and waits for them to complete. We have no control over how many threads are actually used for the execution, because we don't control the thread pool.

Reply:
Will replace with `Parallel.ForEach`

Reply:
Decided to keep it because `Parallel.ForEach` is not async.